interfaces/acados_c/sim_interface.h

```diff
@@ -30,6 +30,7 @@ typedef enum {
     ERK,
     LIFTED_IRK,
     IRK,
+    GNSF,
     NEW_LIFTED_IRK
 } sim_solver_t;
 
```

Comment:
Try to fix the following warnings:

/Users/acados/acados/sim/sim_gnsf.c:219:9: warning: unused variable 'nuhat' [-Wunused-variable]
    int nuhat = dims->nuhat;
        ^
/Users/acados/acados/sim/sim_gnsf.c:524:9: warning: unused variable 'nx' [-Wunused-variable]
    int nx = dims->nx;
        ^
/Users/acados/acados/sim/sim_gnsf.c:591:13: warning: 12 enumeration values not handled in switch: 'EXPL_ODE_FUN',
      'EXPL_ODE_JAC', 'EXPL_ODE_HES'... [-Wswitch]
    switch (fun_type)
            ^
/Users/acados/acados/sim/sim_gnsf.c:591:13: note: add missing switch cases
    switch (fun_type)
            ^
/Users/acados/acados/sim/sim_gnsf.c:622:9: warning: unused variable 'nuhat' [-Wunused-variable]
    int nuhat      = dims->nuhat;
        ^
/Users/acados/acados/sim/sim_gnsf.c:689:9: warning: unused variable 'nuhat' [-Wunused-variable]
    int nuhat      = dims->nuhat;
        ^
/Users/acados/acados/sim/sim_gnsf.c:1016:12: warning: unused variable 'precomputation_time' [-Wunused-variable]
    double precomputation_time = acados_toc(&atimer) * 1000;
           ^
/Users/acados/acados/sim/sim_gnsf.c:1226:11: warning: unused variable 'sim_dim' [-Wunused-variable]
    void *sim_dim = (void *) dims;

Reply:
all these warnings are fixed